pkgs/development/haskell-modules/generic-builder.nix

```diff
@@ -694,6 +713,15 @@ stdenv.mkDerivation ({
     mkdir -p "$installIntermediatesDir"
     cp -r dist/build "$installIntermediatesDir"
     runHook postInstallIntermediates
+
+    ${optionalString jsexe.shouldSymlink ''
+      if [ -d ${binDir} ]; then
+        for exeDir in $installIntermediatesDir/build/*; do
+          exe=$(basename $exeDir)
+          (cd ${binDir} && ln -s $exeDir/$exe.jsexe)
```

Comment:
Note in the sample output this is pointing to inside the same nix store path. 
Is it worth instead using something like `../../$installIntermediatesDir/build/$exe/$exe.jsexe` when `(enableSeparateBinOutput || enableSeparateIntermediatesOutput) == false` to save space in the symlink path ?